openslides_backend/action/generics.py

```diff
@@ -227,40 +226,28 @@ def delete_action_prepare_dataset(self, payload: ActionPayload) -> DataSet:
                         mapped_fields=[field_name],
                         lock_result=True,
                     )
-                    if field.on_delete == OnDelete.PROTECT:
-                        if db_instance.get(field_name):
-                            raise ActionException(
-                                f"You can not delete {self.model} with id {instance['id']}, "
-                                f"because you have to delete the related {str(field.to)} first."
-                            )
-                    elif field.on_delete == OnDelete.CASCADE:
-                        # extract all foreign keys as fqids from the model
-                        foreign_fqids = db_instance.get(field_name, [])
-                        if not isinstance(foreign_fqids, list):
-                            foreign_fqids = [foreign_fqids]
-                        if not isinstance(field, BaseGenericRelationField):
-                            assert not isinstance(field.to, list)
-                            foreign_fqids = [
-                                FullQualifiedId(field.to, id) for id in foreign_fqids
-                            ]
-                        # execute the delete action for all fqids
-                        for fqid in foreign_fqids:
-                            delete_action_class = actions_map.get(
-                                f"{str(fqid.collection)}.delete"
-                            )
-                            if not delete_action_class:
-                                raise ActionException(
-                                    f"Can't cascade the delete action to {str(fqid.collection)} "
-                                    "since no delete action was found."
-                                )
-                            delete_action = delete_action_class(
-                                self.permission, self.database
-                            )
-                            # assume that the delete action uses the standard payload
-                            payload = [{"id": fqid.id}]
-                            self.additional_write_requests.extend(
-                                delete_action.perform(payload, self.user_id)
-                            )
+                    if db_instance.get(field_name):
```

Comment:
I think we have a merge conflict here which @reiterl can not know. I will have a look at it.